packages/realm/src/app-services/App.ts

```diff
@@ -89,6 +85,40 @@ const appByUserId = new Map<string, App<any, any>>();
  * ```
  */
 export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType = Record<string, unknown>> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private static appById = new Map<string, App<any, any>>();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private static appByUserId = new Map<string, App<any, any>>();
```

Comment:
Either we should keep the
```
// TODO: Ensure this doesn't leak
```
Or this should be refactored to use the `binding.WeakRef`.

Reply:
Either we should keep the
```
// TODO: Ensure this doesn't leak
```

Or this should be refactored to use the `binding.WeakRef`:

```suggestion
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private static appById = new Map<string, binding.WeakRef<App<any, any>>>();
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private static appByUserId = new Map<string, binding.WeakRef<App<any, any>>>();
```